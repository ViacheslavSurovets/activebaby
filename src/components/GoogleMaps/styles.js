import styled, { css } from 'styled-components';

const stylesInfoWindow = css`
  font-size: 2rem;
`;

export const InfoWindowTitle = styled.div`
`;

export const InfoWindowTitleText = styled.span`
  ${ stylesInfoWindow };
  &:first-child {
  color: ${ ( { theme } ) => theme.blue };
  }
  &:last-child{
  color: ${ ( { theme } ) => theme.pink };
  }
`;
