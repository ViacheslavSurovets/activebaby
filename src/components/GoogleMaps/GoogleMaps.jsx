import React, { useState } from 'react';
import { GoogleMap, withScriptjs, withGoogleMap, Marker, InfoWindow } from 'react-google-maps';
import styled from 'styled-components';
import { tablet, mobile } from '@utils/media';
import { useTranslation } from 'react-i18next';
import { InfoWindowTitleText, InfoWindowTitle } from './styles';


const Map = () => {
  const [ infoWindow, setInfoWindow ] = useState ( false );
  const { t } = useTranslation ();
  return (
    <GoogleMap
      defaultZoom={ 15 }
      defaultCenter={ { lat: 53.706212, lng: 23.857039 } }>
      <Marker position={ { lat: 53.706212, lng: 23.857039 } } onClick={ () => setInfoWindow ( true ) } />
      { infoWindow &&
      <InfoWindow position={ { lat: 53.706212, lng: 23.857039 } } onCloseClick={ () => setInfoWindow ( false ) }>
        <>
          <InfoWindowTitle>
            <InfoWindowTitleText>Active</InfoWindowTitleText>
            <InfoWindowTitleText>Baby</InfoWindowTitleText>
          </InfoWindowTitle>
          <p>{ t ( 'googleMaps.tooltipInfo' ) }</p>
        </>
      </InfoWindow> }

    </GoogleMap>
  );
};

const ContainerElement = styled.div`
  width: 90vw;
  height: 50vw;
  ${tablet}{
    height: 40rem;
  }
  ${ mobile }{
  width: 95%;
  height: 40rem;
  };
`;

const WrappedMap = withScriptjs ( withGoogleMap ( Map ) );
const key = process.env.GOOGLE_SECRET;
const GoogleMaps = () => (
  <WrappedMap
    googleMapURL={ `https://maps.googleapis.com/maps/api/js?v=3.exp&libraries=geometry,drawing,places&key=${ key }` }
    containerElement={ <ContainerElement /> }
    mapElement={ <div style={ { height: '100%' } } /> }
    loadingElement={ <div style={ { height: '100%' } } /> }
  />
);

export default GoogleMaps;
