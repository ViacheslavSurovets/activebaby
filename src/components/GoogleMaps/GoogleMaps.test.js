import React from 'react';
import { shallow } from 'enzyme';
import GoogleMaps from './GoogleMaps';


describe ( 'GoogleMaps', () => {
  it ( 'GoogleMaps Snapshot', () => {
    expect ( shallow ( <GoogleMaps /> ) ).toMatchSnapshot ();
  } );
} );
