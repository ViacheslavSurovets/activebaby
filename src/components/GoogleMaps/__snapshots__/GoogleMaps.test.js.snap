// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleMaps GoogleMaps Snapshot 1`] = `ShallowWrapper {}`;
